### ISSUE TITLE
[Apex] switch statements do not contribute to Cyclomatic Complexity numbers

### ISSUE THREAD

**Issue by aidan-harding** [pmd/pmd#4863]
I'm happy to fork the repo and have a go at fixing this if you're happy to accept the changes once you're happy with them. 

**Affects PMD Version:** 6.55

**Rule:**

Cyclomatic Complexity
<https://pmd.github.io/pmd/pmd_rules_java_design.html#cyclomaticcomplexity>

**Description:**

The cyclomatic complexity of a method should be increased by adding a `switch` statement, but it does not. 

**Code Sample demonstrating the issue:**

```apex
public class CyclomaticComplexity {
    void foo() {
        switch on Trigger.operationType {
            when BEFORE_INSERT {
                System.debug('Before Insert');
            }
            when AFTER_INSERT {
                System.debug('After Insert');
            }
        }
    }
}
```

**Expected outcome:**

PMD should report a cyclomatic complexity of 3 (starts at one, then +1 for each branch in the switch). It actually reports a cyclomatic complexity of 1. 

**Running PMD through:** *CLI*

**Comment by aidan-harding** [comment]
I checked this in 7.0.0-rc4 and it has the same behaviour.
